.github/workflows/ci.yml

```diff
@@ -36,7 +36,7 @@ jobs:
           fetch-depth: 10
 
       - name: checkout scala/scala
-        if: ${{ runner.os != 'Windows' }}
+        if: ${{ runner.os == 'Linux' }}
```

Comment:
Not exactly related to this but I realized we are cloning this on mac as well, but not running the smoke tests on mac. So I just switched this back to only clone on the os that we actually run it on.

Reply:
As a separate point, I think we can extract the Linux workflow as a separate thing, and on Mac/Windows just do `npm run build && npm run test`.

To declutter all those if conditionals.

Reply:
I can try to follow-up with another pr and clean this up.